libs/server/Storage/Session/ObjectStore/SortedSetGeoOps.cs

```diff
@@ -41,5 +44,160 @@ public GarnetStatus GeoCommands<TObjectContext>(byte[] key, ref ObjectInput inpu
           where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
             => ReadObjectStoreOperationWithOutput(key, ref input, ref objectContext, ref outputFooter);
 
+        /// <summary>
+        /// Geospatial search and store in destination key.
+        /// </summary>
+        /// <typeparam name="TObjectContext"></typeparam>
+        /// <param name="key"></param>
+        /// <param name="input"></param>
+        /// <param name="output"></param>
+        /// <param name="objectContext"></param>
+        /// <returns></returns>
+        public unsafe GarnetStatus GeoSearchStore<TObjectContext>(ArgSlice key, ArgSlice destination, ref ObjectInput input, ref SpanByteAndMemory output, ref TObjectContext objectContext)
+          where TObjectContext : ITsavoriteContext<byte[], IGarnetObject, ObjectInput, GarnetObjectStoreOutput, long, ObjectSessionFunctions, ObjectStoreFunctions, ObjectStoreAllocator>
+        {
+            var createTransaction = false;
+
+            if (txnManager.state != TxnState.Running)
+            {
+                Debug.Assert(txnManager.state == TxnState.None);
+                createTransaction = true;
+                txnManager.SaveKeyEntryToLock(destination, true, LockType.Exclusive);
+                txnManager.SaveKeyEntryToLock(key, true, LockType.Shared);
+                _ = txnManager.Run(true);
+            }
+            var objectStoreLockableContext = txnManager.ObjectStoreLockableContext;
+
+            var isMemory = false;
+            MemoryHandle ptrHandle = default;
+            var ptr = output.SpanByte.ToPointer();
+            var curr = ptr;
+            var end = curr + output.Length;
+
+            try
+            {
+                var isStoreDist = false;
+                Span<ArgSlice> geoSearchParseState = stackalloc ArgSlice[input.parseState.Count + 1]; // Current MaxSize of this stactalloc will be 144 bytes ( 12 (Max Args) x 12 (sizeof ArgSlice) )
```

Comment:
Why are we regenerating the parse state here?
Also, STOREDIST can only appear in the last argument position

Reply:
Sadly its not, it can appear in middle, that is the first thing I tested it
![image](https://github.com/user-attachments/assets/bee39dcc-f485-4850-91d0-f6ed842c17a6)

Reply:
Hmmm ok so I think the last tokens are in this group ([ASC|DESC],[COUNT count [ANY]],[STOREDIST]) but can be at any order...

Reply:
I have changed the implementation not to use additional stackalloc